src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetInstaller.cs

```diff
@@ -32,16 +39,46 @@ public Task<bool> CanInstallAsync(InstallRequest installationRequest)
             return Task.FromResult(!string.IsNullOrWhiteSpace(installationRequest.Identifier) && !Directory.Exists(installationRequest.Identifier));
         }
 
-        public Task<InstallResult> InstallAsync(InstallRequest installRequest)
+        public async Task<InstallResult> InstallAsync(InstallRequest installRequest)
```

Comment:
I really like design of this method with different catches for different exceptions and conversion into enum values.